src/components/sbb-tag-group/sbb-tag-group.tsx

```diff
@@ -107,44 +112,61 @@ export class SbbTagGroup implements ComponentInterface {
     }
   }
 
-  public connectedCallback(): void {
+  public override connectedCallback(): void {
+    super.connectedCallback();
+    const signal = this._abort.signal;
+    this.addEventListener(
+      'state-change',
+      (e: CustomEvent<TagStateChange>) => this._handleStateChange(e),
+      {
+        signal,
+        passive: true,
+      },
+    );
     if (this.value) {
-      this.valueChanged(this.value);
+      this._valueChanged(this.value);
     }
   }
 
-  private get _tags(): HTMLSbbTagElement[] {
-    return Array.from(this._element.querySelectorAll('sbb-tag')) as HTMLSbbTagElement[];
+  private get _tags(): SbbTag[] {
+    return Array.from(this.querySelectorAll('sbb-tag')) as SbbTag[];
   }
 
-  public render(): JSX.Element {
+  protected override render(): TemplateResult {
     this._tags.forEach((tag, index) => tag.setAttribute('slot', `tag-${index}`));
-    return (
-      <Host role={this.listAccessibilityLabel ? '' : 'group'}>
-        <div class="sbb-tag-group">
-          <ul class="sbb-tag-group__list" aria-label={this.listAccessibilityLabel}>
-            {this._tags.map((_, index) => (
-              <li class="sbb-tag-group__list-item">
+
+    setAttribute(this, 'role', this.listAccessibilityLabel ? '' : 'group');
+
+    return html`
+      <div class="sbb-tag-group">
+        <ul class="sbb-tag-group__list" aria-label=${this.listAccessibilityLabel ?? nothing}>
+          ${this._tags.map(
+            (_, index) =>
+              html`<li class="sbb-tag-group__list-item">
                 <slot
-                  name={`tag-${index}`}
-                  onSlotchange={(): void => {
+                  name=${`tag-${index}`}
+                  @slotchange=${(): void => {
                     this._ensureOnlyOneTagSelected();
-                    this._updateValueByReadingTags();
```

Comment:
Why can we remove that?

Reply:
I think it would be called an unnecessary number of times. It's already called by the `_handleStateChange()` callback of the 'state-change' which now, as opposed to before I think, will be called at init and when the slot changes thanks to the `willUpdate` callback in the `tag` component which behaves differently from an `@Watch`. At least that's my thought, what do you think?

Reply:
I'll analyze it deeper :)